source/SkiaSharp.Views.Uno/SkiaSharp.Views.Uno.WinUI.Wasm/WasmScripts/SkiaSharp.Views.Uno.Wasm.js

```diff
@@ -15,12 +17,37 @@
                     if (!ctx)
                         return false;
 
-                    var buffer = new Uint8ClampedArray(Module.HEAPU8.buffer, pData, width * height * 4);
-                    var imageData = new ImageData(buffer, width, height);
-                    ctx.putImageData(imageData, 0, 0);
+                    var byteLength = width * height * 4;
+
+                    if (isSecureContext) {
+                        // In a secure context (e.g. with threading enabled), creating a view
+                        // from Module.HEAPU8.buffer is not supported, so we're making an
+                        // explicit copy of the wasm memory.
```

Comment:
How expensive is this? It is now a double copy? First the copy and then the copy to canvas?